pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -107,16 +114,25 @@ private TransferableBlock produceSortedBlock() {
     }
 
     if (!_isSortedBlockConstructed) {
-      LinkedList<Object[]> rows = new LinkedList<>();
-      while (_rows.size() > _offset) {
-        Object[] row = _rows.poll();
-        rows.addFirst(row);
-      }
       _isSortedBlockConstructed = true;
-      if (rows.size() == 0) {
-        return TransferableBlockUtils.getEndOfStreamTransferableBlock();
+      if (_priorityQueue == null) {
```

Comment:
Not related to changes in this PR but just wondering if `_isSortedBlockConstructed` be volatile ?

Reply:
it doesn't need to be as the entire operator is single-threaded.